### DISCUSSION TITLE
Task Management: Possibility to choose Tasks in the sheet "Weekly hours"

### DISCUSSION THREAD

**Discussion by CHolzbaur** [kimai/Paid-plugins#209] [Questions & Answers]
Hello,

In our administration, we currently use Kimai with the Start/Stop method for time tracking. For our production team, we plan to enter the hours for the entire week into the "weekly hours" sheet on Friday evening.

To calculate the times correcly, we have to assign the correct task (red arrow in the attachment) to the timesheet for proper tracking. Is it possible to configure this in Kimai, or is there an alternative way to achieve this?

Thank you for your help!
Best regards
CHolzbaur

![Task](https://github.com/user-attachments/assets/309c4b9c-bbb9-4f23-a422-ac400af618d3)

**Comment by kevinpapst** [comment]
Not possible.
I could probably try to add some logic that magically attaches a task to a timesheet upon saving. 
But adding a selector on that screen is not on the roadmap.
Actually this screen is (as of now) not extendable at all.

**Reply by CHolzbaur** [reply]
Thank you for your feedback.

This "magical solution" sounds interesting, but is not what I am searching for.

As a task already has a defined project and activity, the extension of this screen would even be overkill. It would be "sufficient" to have only a screen similar to the one from the first post but only with one selector "task" instead of the current two selectors. The rest would be the same.

As far as I can oversee the database, the difference between a **time-sheet with task** to a **time-sheet without task** is a one input in the table `kimai2_task_timesheets`. 

Is it possible to copy the code of the existing screen and modify it in such a way?
It would do mostly the same as right now, but:
-  Instead of expecting the userinput for **"project"** and **"activity"** - there would only be one selector for **"task"** 
- The code would gather the project and task from the database: ```SELECT `activity_id`, `project_id` FROM `kimai2_tasks` WHERE ...```
- After creating the time-sheet it would link the new time-sheet ```INSERT INTO `kimai2_task_timesheets` .... ```

I would buy you a beer or a coffee.
